climada_petals/entity/exposures/openstreetmap/osm_dataloader.py

```diff
@@ -0,0 +1,945 @@
+                               substance='waterwaste' or 
+                               substance='wastewater' or
+                               (natural='water' and water='wastewater') or
+                               man_made='wastewater_plant' or
+                               man_made='wastewater_tank' or
+                               utility='sewerage'"""},
+         'food' : {
+             'osm_keys' : ['shop','name'],
+             'osm_query' : """shop='supermarket' or shop='greengrocer' or
+                              shop='grocery' or shop='general' or 
+                              shop='bakery'"""},
+         }
+
+
+class OSMRaw:
+    """
+    functions to obtain entire raw datasets from OSM, 
+    from different sources"""
+
+    def __init__(self):
+        self.geofabrik_url = 'https://download.geofabrik.de/'
+        self.planet_url = 'https://planet.openstreetmap.org/pbf/planet-latest.osm.pbf'
+
+    def _create_gf_download_url(self, iso3, file_format):
+        """
+        create string with download-api from geofabrik
+        
+        Parameters
+        ----------
+        iso3 : str
+            ISO3 code of country to download
+        file_format : str
+            Format in which file should be downloaded; ESRI Shapefiles ('shp') 
+            or osm-Protocolbuffer Binary Format ('pbf')
+        
+        Returns
+        -------
+        str : Geofabrik ownload-api for the requested country.
+        
+        See also
+        --------
+        DICT_GEOFABRIK for exceptions / special regions.
+        """
+
+        if file_format == 'shp':
+            return f'{self.geofabrik_url}{DICT_GEOFABRIK[iso3][0]}/{DICT_GEOFABRIK[iso3][1]}-latest-free.shp.zip'
+        elif file_format == 'pbf':
+            return f'{self.geofabrik_url}{DICT_GEOFABRIK[iso3][0]}/{DICT_GEOFABRIK[iso3][1]}-latest.osm.pbf'
+        else:
+            LOGGER.error('invalid file format. Please choose one of [shp, pbf]')
+
+    def get_data_geofabrik(self, iso3, file_format='pbf', save_path=DATA_DIR):
+        """
+        Download country files with all OSM map info from the provider 
+        Geofabrik.de, if doesn't exist, yet.
+        
+        Parameters
+        ----------
+        iso3 : str
+            ISO3 code of country to download
+            Exceptions: Russia is divided into European and Asian part 
+            ('RUS-E', 'RUS-A'), Canary Islands are 'IC'.
+        file_format : str
+            Format in which file should be downloaded; options are 
+            ESRI Shapefiles (shp), which can easily be loaded into gdfs,
+            or osm-Protocolbuffer Binary Format (pbf), which is smaller in 
+            size, but has a more complicated query syntax to load (functions
+            are provided in the OSMFileQuery class).
+        save_path : str or pathlib.Path
+            Folder in which to save the file
+            
+        Returns
```

Comment:
It might be worth returning the file path here. The user won't always know the downloaded file's name and this might speed up someone's workflow. Same applies to other methods in this class.

(Or to save time we could leave this as a GitHub Issue.)

Reply:
It is logging the path, tough.  LOGGER.info(f'Downloading file as {local_filepath}') includes the complete file name.

Reply:
i'm ok with that. and if for some reason i need the filepath in a script i can get the name from `DICT_GEOFABRIK`